core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -828,16 +830,17 @@ public FileInfo getFileInfo(AlluxioURI path, GetStatusContext context)
         FileSystemMasterAuditContext auditContext =
             createAuditContext("getFileInfo", path, null, null)) {
 
-      if (syncMetadata(rpcContext, path, context.getOptions().getCommonOptions(),
+      if (!syncMetadata(rpcContext, path, context.getOptions().getCommonOptions(),
           DescendantType.ONE, auditContext, LockedInodePath::getInodeOrNull,
           (inodePath, permChecker) -> permChecker.checkPermission(Mode.Bits.READ, inodePath),
-          true)) {
+          true).equals(NOT_NEEDED)) {
```

Comment:
Has this condition been modified? It seems like it would be equivalent to `.equals(OK)`?

Reply:
The rationale is that if a sync had failed, loadMetadata would  likely fail in similar fashion since the code path is very similar. So we can save some unnecessary loadMetadata operations?